stix2validator/shoulds.py

```diff
@@ -274,14 +274,16 @@ def relationships_strict(instance):
         return
 
     r_type = instance['relationship_type']
-    r_source = re.search("(.+)\-\-", instance['source_ref']).group(1)
-    r_target = re.search("(.+)\-\-", instance['target_ref']).group(1)
+    try:
+        r_source = re.search("(.+)\-\-", instance['source_ref']).group(1)
+        r_target = re.search("(.+)\-\-", instance['target_ref']).group(1)
+    except (AttributeError, TypeError):
+        return  # Schemas already catch these
 
     if (r_type in enums.COMMON_RELATIONSHIPS or
             r_source in enums.NON_SDOS or
             r_target in enums.NON_SDOS):
-        # Schemas will already catch relationships not allowed by spec
-        return
+        return  # Schemas already catch invalid relationships
```

Comment:
Is this just to short-circuit the tests below? Or are we trying to prevent two error messages (one from the schema validator and one from our code)? Would it be easier to just have a list (set?) of `(source, type, target)` tuples? 

_Just thinking out loud here. Don't mind me._ 😁